// web/src/components/ProductCard.tsx
import { Product } from "../types/product";
import { motion } from "framer-motion";

interface ProductCardProps {
  product: Product;
  isLowest?: boolean;
  savings?: number;
  onClick?: () => void;
  onToggleWishlist?: () => void;
  isWishlisted?: boolean;
}

export default function ProductCard({
  product,
  isLowest,
  savings,
  onClick,
  onToggleWishlist,
  isWishlisted
}: ProductCardProps) {
  
  // Custom glowing badge colors based on e-commerce platform
  const getPlatformStyle = (platform: string) => {
    const p = platform.toLowerCase();
    if (p.includes('amazon')) return 'bg-orange-500/20 text-orange-400 border-orange-500/30';
    if (p.includes('flipkart')) return 'bg-blue-500/20 text-blue-400 border-blue-500/30';
    if (p.includes('ebay')) return 'bg-green-500/20 text-green-400 border-green-500/30';
    if (p.includes('walmart')) return 'bg-sky-500/20 text-sky-400 border-sky-500/30';
    if (p.includes('google')) return 'bg-red-500/20 text-red-400 border-red-500/30';
    if (p.includes('best buy')) return 'bg-yellow-500/20 text-yellow-400 border-yellow-500/30';
    if (p.includes('snapdeal')) return 'bg-rose-500/20 text-rose-400 border-rose-500/30';
    return 'bg-cyan-500/20 text-cyan-400 border-cyan-500/30'; // Default Neon Cyan
  };

  const platformStyle = getPlatformStyle(product.platform);

  return (
    <motion.div 
      whileHover={{ y: -8, scale: 1.02 }}
      transition={{ type: "spring", stiffness: 300, damping: 20 }}
      className={`relative glass-card rounded-3xl overflow-hidden border flex flex-col h-full transition-shadow duration-300 ${isLowest ? 'border-cyan-400 shadow-[0_0_30px_rgba(34,211,238,0.2)]' : 'border-slate-800 hover:border-slate-600 hover:shadow-[0_10px_30px_rgba(0,0,0,0.5)]'}`}
    >      
      
      {/* Top Left Badges (Best Value) */}
      <div className="absolute top-3 left-3 z-30 flex flex-col gap-2">
        {isLowest && (
          <div className="bg-gradient-to-r from-cyan-500 to-blue-600 text-white text-xs font-bold px-3 py-1.5 rounded-full shadow-lg shadow-cyan-500/30 flex items-center gap-1 backdrop-blur-md border border-cyan-400/50 animate-pulse">
            <span>🏆</span> Best Value
          </div>
        )}
      </div>

      {/* Top Right Wishlist Button */}
      <button 
        onClick={(e) => { e.stopPropagation(); onToggleWishlist?.(); }}
        className="absolute top-3 right-3 z-10 bg-slate-900/60 backdrop-blur-md p-2 rounded-full border border-slate-700 hover:scale-110 transition-transform shadow-lg"
      >
        {isWishlisted ? '💖' : '🤍'}
      </button>

      {/* Image Container with Dark Vignette */}
      <div 
        onClick={onClick}
        className="relative h-56 bg-slate-900 overflow-hidden group cursor-pointer p-6 flex items-center justify-center border-b border-slate-800"
      >
        <div className="absolute inset-0 bg-gradient-to-t from-slate-900 via-transparent to-transparent z-0 opacity-80"></div>
        <img
          src={product.image}
          alt={product.title}
          className="max-w-full max-h-full object-contain relative z-10 group-hover:scale-110 transition-transform duration-500 drop-shadow-2xl"
        />
      </div>

      {/* Content Section */}
      <div className="p-5 flex-1 flex flex-col bg-slate-900/80 backdrop-blur-xl">
        
        {/* Badges Row */}
        <div className="flex items-center justify-between mb-3">
          <span className={`text-xs font-bold px-2.5 py-1 rounded-md border ${platformStyle}`}>
            {product.platform}
          </span>

          {product.rating ? (
            <span className="flex items-center text-yellow-400 text-xs font-semibold bg-yellow-400/10 px-2 py-1 rounded border border-yellow-400/20">
              <span className="mr-1">⭐</span> {product.rating}
            </span>
          ) : null}
        </div>

        {/* Product Title */}
        <h3 onClick={onClick} className="text-slate-200 text-sm font-medium leading-relaxed line-clamp-2 cursor-pointer hover:text-cyan-400 transition-colors flex-1">
          {product.title}
        </h3>

        {/* Pricing & Savings */}
        <div className="mt-4 pt-4 border-t border-slate-800">
          <p className="text-3xl font-bold bg-gradient-to-r from-white to-slate-400 bg-clip-text text-transparent flex items-baseline gap-1">
            <span className="text-xl text-cyan-400 font-medium">₹</span>
            {product.price.toLocaleString()}
          </p>

          {isLowest && savings && savings > 0 ? (
            <p className="text-emerald-400 text-xs mt-2 font-medium flex items-center gap-1 bg-emerald-400/10 w-fit px-2.5 py-1 rounded-md border border-emerald-400/20">
               ↓ Save ₹{(savings).toLocaleString()} vs highest
            </p>
          ) : (
             <div className="h-7 mt-2"></div>
          )}
        </div>

        {/* Action Button */}
        <a
          href={product.url}
          target="_blank"
          rel="noreferrer"
          className="mt-5 flex items-center justify-center w-full bg-slate-800 hover:bg-gradient-to-r hover:from-cyan-500 hover:to-blue-600 text-white py-3 rounded-xl font-semibold transition-all duration-300 group border border-slate-700 hover:border-cyan-400 hover:shadow-[0_0_20px_rgba(6,182,212,0.4)]"
        >
          <span>View on {product.platform}</span>
          <span className="ml-2 group-hover:translate-x-1 transition-transform">→</span>
        </a>
      </div>
    </motion.div>
  );
}
